android/app/src/main/java/org/electroncash/electroncash3/ColdLoad.kt

```diff
@@ -69,4 +134,72 @@ class ColdLoadDialog : AlertDialogFragment() {
             dismiss()
         } catch (e: ToastException) { e.show() }
     }
+
+
+    /**
+     * Check if a loaded transaction is signed.
+     * Displays the signing status below the raw TX field.
+     * (signed, partially signed, empty or invalid)
+     */
+    private fun updateStatusText(tx: PyObject) {
+        try {
+            val txInfo = daemonModel.wallet!!.callAttr("get_tx_info", tx)
+
+            // Check if the transaction can be processed by this wallet or not
+            if (txInfo["amount"] == null) {
+                idTxStatus.setText(R.string.transaction_unrelated)
+            } else {
+                idTxStatus.setText(txInfo["status"].toString())
+            }
+        } catch (e: PyException) {
+            idTxStatus.setText(R.string.invalid)
+        }
+    }
+}
+
+/* Check if the wallet can sign the transaction */
+fun canSign(tx: PyObject): Boolean {
+    return try {
+        !tx.callAttr("is_complete").toBoolean() and
+                daemonModel.wallet!!.callAttr("can_sign", tx).toBoolean()
+    } catch (e: PyException) {
+        false
+    }
+}
+
+/* Check if the transaction is ready to be broadcasted */
+fun canBroadcast(tx: PyObject): Boolean {
+    return try {
+        tx.callAttr("is_complete").toBoolean()
+    } catch (e: PyException) {
+        false
+    }
+}
+
+/**
+ * Sign a loaded transaction dialog.
+ */
+class SignPasswordDialog : PasswordDialog<Unit>() {
+
+    val coldLoadDialog by lazy { targetFragment as ColdLoadDialog }
+    val signSchnorr = daemonModel.walletType == "standard"
```

Comment:
Standard and multisig aren't the only wallet types: see comment in Transactions.kt.